src/interface/sparse_pack_base.cpp

```diff
@@ -85,21 +85,22 @@ SparsePackBase SparsePackBase::Build(T *pmd, const PackDescriptor &desc) {
   pack.with_fluxes_ = desc.with_fluxes;
   pack.coarse_ = desc.coarse;
   pack.nvar_ = desc.vars.size();
+  pack.flat_ = desc.flat;
 
   // Count up the size of the array that is required
   int max_size = 0;
-  int nblocks = 0;
-  int ndim = 3;
+  int nblocks = desc.flat;
+  int size = 0;
   ForEachBlock(pmd, [&](int b, mbd_t *pmbd) {
-    int size = 0;
-    nblocks++;
+    if (!desc.flat) {
+      size = 0;
+      nblocks++;
+    }
     for (auto &pv : pmbd->GetCellVariableVector()) {
       for (int i = 0; i < nvar; ++i) {
         if (desc.IncludeVariable(i, pv)) {
           if (pv->IsAllocated()) {
             size += pv->GetDim(6) * pv->GetDim(5) * pv->GetDim(4);
-            ndim = (pv->GetDim(1) > 1 ? 1 : 0) + (pv->GetDim(2) > 1 ? 1 : 0) +
-                   (pv->GetDim(3) > 1 ? 1 : 0);
```

Comment:
Removed since, after discussing with @lroberts36, we agreed this logic isn't actually needed.